### ISSUE TITLE
Add respond_to? and respond_to_missing? (Currently doesn't work with Object#try)

### ISSUE THREAD

**Issue by kenn** [lassebunk/dish#9]
Dish causes behavioral difference on [Object#try](http://api.rubyonrails.org/classes/Object.html#method-i-try)

``` ruby
require 'dish'

dish = Dish({a:1,b:2})
 => #<Dish::Plate:0x007fcfc2c99070 @_original_hash={"a"=>1, "b"=>2}>

dish.a
 => 1

dish.try(:a)
 => nil
```

I haven't read the source, but is this expected? Workarounds?

**Comment by lassebunk** [comment]
Hi Kenn,

Thanks for writing.
No, this is unexpected. We will fix this.

Thanks.

**Comment by lassebunk** [comment]
The problem is [this line](https://github.com/rails/rails/blob/b4fa6516c1fc81f7da2a92adee9634d2771b7df0/activesupport/lib/active_support/core_ext/object/try.rb#L45).
We are fixing the `respond_to?` method so this will work.

**Comment by kenn** [comment]
Yeah that does sound reasonable. Also defining `respond_to_missing?` may be a good idea. http://robots.thoughtbot.com/always-define-respond-to-missing-when-overriding

**Comment by lassebunk** [comment]
Yep. Thanks.

Den lørdag den 8. marts 2014 skrev Kenn Ejima notifications@github.com:

**Comment by malkomalko** [comment]
Any updates on this?

**Comment by lassebunk** [comment]
Not yet. It's coming though.

**Comment by MarkusHarmsen** [comment]
Should be fixed since 9ccc082.

**Comment by lassebunk** [comment]
:+1:

**Comment by lassebunk** [comment]
Now available in v0.0.5. Thanks :+1:
